csp_billing_adapter_local/plugin.py

```diff
@@ -96,6 +97,31 @@ def update_cache(config: Config, cache: dict, replace: bool):
         json.dump(cache, f)
 
 
+@csp_billing_adapter.hookimpl(trylast=True)
+def save_metering_archive(config: Config, archive_data: list):
+    """Update local storage archive with new content"""
+    with open(get_local_path(ARCHIVE_FILE), 'w', encoding='utf-8') as f:
+        json.dump(archive_data, f)
```

Comment:
If the system goes down in the middle of the `dump` operation we generate a corrupted archive, which is handled in the reading. However, we also loose the data since the file is opened with `w`. I think it would make sense to create a backup and unlink the backup if the write is successful. The backup will be 1 billing cycle short but that is better than loosing an undefined amount of data, IMHO.

Reply:
Noted: will merge this and create a separate PR